packages/plugin-babel/src/plugin.ts

```diff
@@ -66,6 +68,16 @@ export const getDefaultBabelOptions = (
         require.resolve('@babel/plugin-proposal-decorators'),
         config.source.decorators,
       ],
+      // If you are using @babel/preset-env and legacy decorators, you must ensure the class elements transform is enabled regardless of your targets, because Babel only supports compiling legacy decorators when also compiling class properties:
```

Comment:
If the plugin user is using `@babel/preset-env`, will this config conflict with the user config?

Reply:
I think it won't
1. pluginBabel seems to be only used in rspack mode and at present it is rarely used with babel-preset.

2. For webpack, I haven't added this plugin, get error in legacy decorators
https://github.com/web-infra-dev/rsbuild/blob/b4355beee8753b621f5ad0f5434f8e85a91f8d15/packages/babel-preset/src/base.ts#L48

Reply:
> at present it is rarely used with babel-preset.

Yes, it's not a common usage, but it's better if it's not broken by this change.

Reply:
babel-preset will add this plugin automatically, I think it is smart enough
I tried use this
```js
 plugins: [
                [
                  '@babel/plugin-proposal-decorators/lib/index.js',
                  {
                    version: 'legacy'
                  }
                ],
                '@babel/plugin-transform-class-properties/lib/index.js'
              ],
              presets: [
                [
                  '@babel/preset-typescript/lib/index.js',
                  {
                    allowNamespaces: true,
                    allExtensions: true,
                    allowDeclareFields: true,
                    optimizeConstEnums: true,
                    isTSX: true
                  }
                ],
                [
                  '@babel/preset-env/lib/index.js',
                  {
                    modules: false,
                    exclude: [
                      'transform-typeof-symbol'
                    ],
                    bugfixes: true,
                    corejs: undefined,
                    targets: [
                      'Chrome >= 53'
                    ],
                    include: [
                      '@babel/plugin-transform-class-properties'
                    ]
                  }
                ]
              ],
```